internal/fnruntime/container.go

```diff
@@ -170,8 +174,21 @@ func (f *ContainerFn) prepareImage() error {
 	}
 	if strings.Contains(string(output), strings.Split(f.Image, ":")[0]) {
 		// image exists locally
+		foundImageInLocalCache = true
+	}
+
+	// If env var PreferLocalCacheEnv is set to "true", we scan the local images
+	// first. If there is a match, we just return. This can be useful for local
+	// development.
+	if strings.ToLower(os.Getenv(PreferLocalCacheEnv)) == "true" && foundImageInLocalCache {
```

Comment:
I am wondering if we should do this:

```
err := docker.ImagePull(img)
if err == nil {
  // all good
 return
}
if errors.Is(err, IsNotExist) {
   err := docker.ImageLs(img)
   // if img is local, we should err == nil
   if err == nil {
     return 
  }
}
return err
```

The benefit of this approach is... we don't have to instruct users about these special cases. Leaking the internal details to end users is not desired.

Reply:
This approach works fine for function consumer. But there is an issue for function developers:
If the user has a image `gcr.io/my-registry/my-img:unstable` in the remote registry, while the user is developing a new feature locally and reuse the image name and tag `gcr.io/my-registry/my-img:unstable`. When the user run `kpt fn render|eval`, there is no way to let kpt use local images when there is a name confliction with the remote registry.

Maybe we can solve it by guiding our users to avoid using a name and tag that exist in a remote registry.

Reply:
hmm...good point. Optimizing UX for function consumers is definitely important, but I also see `function authors` running in to it very easily so that is also poor. Wondering how we can make it easy for function authors to catch this error in case of conflict.  And second, about config naming, keeping it similar to `ImagePullAlways` that we have in k8s might help as well.